### PR TITLE
test(core): add exception cases for role api and scope api

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

Add API response guards and exception integration test cases to the role and scope related APIs. 

## Testing

test locally

## Checklist

- [ ] `.changeset`
- [ ] unit tests
- [x] integration tests
- [ ] docs

OR

- [ ] This PR is not applicable for the checklist

### CODE REVIEW

File: packages/core/src/routes/role.scope.ts [hunk 1]
```diff
@@ -26,6 +27,9 @@ export default function roleScopeRoutes<T extends AuthedRouter>(
     koaPagination({ isOptional: true }),
     koaGuard({
       params: object({ id: string().min(1) }),
+      // @wangsijie FIXME: resource response should be required
+      response: Scopes.guard.extend({ resource: Resources.guard.optional() }).array(),
```
[the remark this answers is not in the document]
Comment (simeng-li) [reply packages/core/src/routes/role.scope.ts:31]: Resolved, per offline discussion, keep it inline for now, consider move the guard to schema if it's type will be referenced by others.

File: packages/core/src/routes/role.ts [hunk 2]
```diff
@@ -41,57 +41,83 @@ export default function roleRoutes<T extends AuthedRouter>(
...
-            return {
-              ...role,
-              usersCount: count,
-              featuredUsers: users.map(({ id, avatar, name }) => ({ id, avatar, name })),
-            };
+  router.get(
+    '/roles',
+    koaPagination(),
+    koaGuard({
+      response: Roles.guard
```
[the remark this answers is not in the document]
Comment (simeng-li) [reply packages/core/src/routes/role.ts:48]: same as above

File: packages/schemas/src/types/user.ts [hunk 3]
```diff
@@ -15,11 +17,26 @@ export const userInfoSelectFields = Object.freeze([
...
 ] as const);
 
-export type UserInfo<Keys extends keyof CreateUser = (typeof userInfoSelectFields)[number]> = Pick<
-  CreateUser,
+export type UserInfo<Keys extends keyof User = (typeof userInfoSelectFields)[number]> = Pick<
+  User,
   Keys
 >;
 
+export const userInfoResponseGuard: CreateGuard<UserInfo> = Users.guard.pick({
```
Comment (wangsijie) [inline packages/schemas/src/types/user.ts:25]: I alse add this in https://github.com/logto-io/logto/pull/3809

**Review by wangsijie — APPROVED** [review]

**Review by darcyYe — APPROVED** [review]

File: packages/schemas/src/types/user.ts (hunk 3 above)
Comment (simeng-li) [reply packages/schemas/src/types/user.ts:25]: racing time！

**Pull request merged**